sdk/src/test/java/com/horizen/storage/StorageTest.java

```diff
@@ -1,72 +1,159 @@
+        List<Pair<ByteArrayWrapper,ByteArrayWrapper>> u1 = storageFixture.getKeyValueList(firstUpdateSize);
+        s.update(version1, u1, new ArrayList<>());
 
         assertFalse("Storage expected to be not empty.", s.isEmpty());
-        assertEquals("Storage must contain 3 items.", 3, s.getAll().size());
+        assertEquals("Storage must contain 3 items.", firstUpdateSize, s.getAll().size());
         assertEquals("Storage must contain value for key - " + u1.get(0).getKey(), u1.get(0).getValue(), s.get(u1.get(0).getKey()).get());
         assertTrue("Storage must contain same elements as sample.", u1.containsAll(s.getAll()));
         assertEquals("Storage must have specified last version.", version1, s.lastVersionID().get());
-        assertEquals("Storage must have only one version.", 1, s.rollbackVersions().size());
-        assertTrue("Storage must have only specified version.", s.rollbackVersions().contains(version1));
+        assertEquals("Storage must have two versions", 2, s.rollbackVersions().size());
+        assertTrue("Storage must have specified versions", s.rollbackVersions().containsAll(Arrays.asList(version0, version1)));
 
         ByteArrayWrapper version2 = storageFixture.getVersion();
         List<Pair<ByteArrayWrapper,ByteArrayWrapper>> u2 = new ArrayList<>();
 
         for (Pair<ByteArrayWrapper,ByteArrayWrapper> i : u1)
-            u2.add(new Pair<ByteArrayWrapper,ByteArrayWrapper>(i.getKey(), storageFixture.getValue()));
+            u2.add(new Pair<>(i.getKey(), storageFixture.getValue()));
 
-        s.update(version2, u2, new ArrayList<ByteArrayWrapper>());
+        s.update(version2, u2, new ArrayList<>());
 
         assertEquals("Storage must contain 3 items.", 3, s.getAll().size());
         assertEquals("Storage must contain value for key - " + u2.get(0).getKey(), u2.get(0).getValue(), s.get(u2.get(0).getKey()).get());
         assertTrue("Storage must contain same elements as sample.", u2.containsAll(s.getAll()));
         assertEquals("Storage must have specified version.", version2, s.lastVersionID().get());
-        assertEquals("Storage must have two versions.", 2, s.rollbackVersions().size());
-        assertTrue("Storage must contain specified version.", s.rollbackVersions().contains(version1));
-        assertTrue("Storage must contain specified version.", s.rollbackVersions().contains(version2));
+        assertEquals("Storage must have three versions.", 3, s.rollbackVersions().size());
+        assertTrue("Storage must have specified versions", s.rollbackVersions().containsAll(Arrays.asList(version0, version1, version2)));
+
 
         ByteArrayWrapper version3 = storageFixture.getVersion();
         List<ByteArrayWrapper> r = new ArrayList<>();
 
         for (Pair<ByteArrayWrapper,ByteArrayWrapper> i : u2.subList(1,3))
             r.add(i.getKey());
 
-        s.update(version3, new ArrayList<Pair<ByteArrayWrapper,ByteArrayWrapper>>(), r);
+        s.update(version3, new ArrayList<>(), r);
```

Comment:
Let's add the test case where we try to remove non-existing elements - should be successful.

Reply:
done